### PR TITLE
Fix sequential version criterion

### LINKED ISSUE #49: Mistake in sequential version number check
https://github.com/JuliaRegistries/General/pull/4722

### PR DESCRIPTION
**Pull request by ericphanson**
Fixes #49 by adapting `check_new_version` from https://github.com/JuliaLang/julia/commit/5741c7f53c5ea443bbd73cc5157b83de415c6d1b as suggested in the issue. I vendored the functions `thispatch` etc in `semvar.jl` as well, since I didn't think they were part of the public interface.

### CODE REVIEW

File: test/automerge-unit.jl [hunk 1]
```diff
@@ -45,32 +45,49 @@ end
...
+        @test !AutoMerge.meets_sequential_version_number([v"0.1.2"], v"0.2.2")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.1"], v"1.1.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.2.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.2.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.3"], v"1.1.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.0.1")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.1.0")[1]
+        @test !AutoMerge.meets_sequential_version_number([v"1.0.0"], v"2.1.0")[1]
+
+        @test AutoMerge.meets_sequential_version_number([v"0.0.1", v"0.1.0"], v"0.0.2")[1]
```
Comment (ericphanson) [inline test/automerge-unit.jl]: I believe this test in particular shows the new functionality. The following tests are adapted from https://github.com/JuliaLang/julia/commit/5741c7f53c5ea443bbd73cc5157b83de415c6d1b.

**Comment by ericphanson** [comment]
Ah, I should bump the version. Is this breaking, or a bugfix? I can kind of see it either way.

**Comment by DilumAluthge** [comment]
> Ah, I should bump the version. Is this breaking, or a bugfix? I can kind of see it either way.

It’s a bug fix. I bumped the patch.

**Comment by DilumAluthge** [comment]
Now I have to figure out how to run the integration tests.

File: src/AutoMerge/guidelines.jl [hunk 2]
```diff
@@ -129,33 +129,45 @@ function meets_repo_url_requirement(pkg::String; registry_head::String)
...
-        return false, "Does not meet sequential version number guideline", :invalid
+        return _invalid_sequential_version("increment is not one of: 0.0.1, 0.1.0, 1.0.0")
     end
 end
 
+function meets_sequential_version_number(existing::Vector{VersionNumber}, ver::VersionNumber)
+    always_assert(!isempty(existing))
+    issorted(existing) || (existing = sort(existing))
+    idx = searchsortedlast(existing, ver)
+    idx > 0 || return invalid("version $ver less than least existing version $(existing[1])")
```
Comment (ericphanson) [inline src/AutoMerge/guidelines.jl]: This “invalid” needs to be updated (sorry, not sure how to do backticks from my phone)

Comment (DilumAluthge) [reply src/AutoMerge/guidelines.jl]: Done (I think).

Are you iPhone or Android? If you are iPhone, click the button in the lower left hand corner that says ".?123". Then, press and hold over the apostrophe, and options for the backtick should pop up.

Comment (ericphanson) [reply src/AutoMerge/guidelines.jl]: Ooh, thanks! `now I can write code :)`

**Comment by DilumAluthge** [comment]
Thank you @ericphanson! FYI if you're interested in contributing further, I'd really like to see us get our code coverage up to 100% (issue #37) because of how important this code is to the General registry.

**Comment by ericphanson** [comment]
No problem! Sure, I’ll try to help add some tests (though maybe won’t get to 100%).

**Pull request merged**